### ISSUE TITLE
Profiling changes column names of original dataframe

### ISSUE THREAD

**Issue by tobycheese** [pandas-profiling/pandas-profiling#243]
**Describe the bug**

profile_report() has the side effect of changing spaces in column names to underscores.

**To Reproduce**
```python
"""
Test for issue XXX:
https://github.com/pandas-profiling/pandas-profiling/issues/XXX
"""
import pandas as pd
import pandas_profiling

def test_issueXXX():
    df = pd.DataFrame(data=[[1,2],[3,4]], columns=['Col 1', 'Col 2'])
    columns_before = df.columns
    df.profile_report()
    assert df.columns.tolist() == columns_before.tolist()
```

**Version information:**

Version information is essential in reproducing and resolving bugs. Please report:

* _Python version_: Python 3.7.4
* _Environment_: PyCharm
* _`pip`_: I am skipping this as the bug is obvious.

**Comment by Mi5ke** [comment]
Same in Python 3.6, every column with spaces in its name is renamed to used underscores in place of spaces. (In my case, that then causes subsequent processing pipeline to fail. Could keep a list of original column names and re-instate them, but would rather profile_report caused no changes.)

**Comment by sbrugman** [comment]
The column names are altered as spaces and colons caused errors for some operations. I agree that `pandas-profiling` should not have side-effects. Either we solve the problems that occur with certain characters or we re-instate the column names afterwards.

**Comment by tobycheese** [comment]
It's not fixed in 2.4.0.
